lib/data/services/firestore/firestore_service.dart

```diff
@@ -0,0 +1,68 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:flutter/material.dart';
+import 'package:get/get.dart';
+import 'package:ns_utils/extensions/map.dart';
+
+import '../../../ui/utils/app_loader.dart';
+import '../../../ui/utils/constant.dart';
+import '../../models/app_response.dart';
+import '../../models/user.dart';
+
+FireStoreService get fireStore => Get.find<FireStoreService>();
+
+class FireStoreService {
+  CollectionReference users = FirebaseFirestore.instance.collection('users');
+  addUser({
+    required User user,
+  }) {
+    // Call the user's CollectionReference to add a new user
+    users
+        .doc(user.email)
+        .set({
+          Constants.displayName: user.displayName,
+          Constants.score: user.score,
+        })
+        .then((value) => debugPrint("User added successfully!"))
+        .catchError((error) => debugPrint('Error adding user $error'));
+  }
+
+  Future<AppResponse> getAllTopScoredUsers() async {
+    try {
+      AppLoader.show();
+      QuerySnapshot querySnapshot = await users.get();
+      List<User> userList = [];
+      List<User> top10UserList = [];
+      for (var doc in querySnapshot.docs) {
+        final data = doc.data() as Map<String, dynamic>;
+
+        userList.add(
```

Comment:
No need of this code

Reply:
Yes removed, after providing an order clause no need for this code